packages/params-middleware/src/middleware.js

```diff
@@ -0,0 +1,45 @@
+// @flow
+
+import { RSAA } from 'redux-api-middleware';
+
+import type { Dispatch } from 'redux';
+import type { Action } from './types';
+
+function parametriseEndpoint({ endpoint, types }) {
+  const params = types[0]?.meta?.params;
+
+  if (typeof endpoint === 'function' || !params) {
+    return endpoint;
+  }
+
+  const searchParams = new URLSearchParams();
+
+  Object.keys(params)
+    .filter(key => !!params[key])
+    .forEach(key => {
+      if (Array.isArray(params[key])) {
+        params[key].forEach(inner => searchParams.append(`${key}[]`, inner));
+      } else {
+        searchParams.append(key, params[key]);
+      }
+    });
+
+  return endpoint + '?' + searchParams.toString();
```

Comment:
Maybe we should check if endpoint has another params already defined (ie. constant params that will never change can be defined directly in action definition https://example.com/contacts?scope=owner)

Reply:
Yeah, it is just a proposal, it can be a future improvement.

Reply:
What do you think about using stringify from https://www.npmjs.com/package/qs?
There are a lot more edge cases to cover and at least nesting is often required.

Reply:
And with regards to endpoint already having a query param, I don't think this is an issue.
Although passing of the params could be simplified, see my other comment.